src/Microsoft.Azure.WebJobs.Host/ExponentialBackoffRetryAttribute.cs

```diff
@@ -49,9 +48,21 @@ public ExponentialBackoffRetryAttribute(int maxRetryCount, string minimumInterva
 
         public override TimeSpan GetNextDelay(RetryContext context)
         {
+            IDelayStrategy delayStrategy;
+
+            if (context.StateDictionary.TryGetValue(nameof(RandomizedExponentialBackoffStrategy), out object stateObject))
```

Comment:
I just want to point out that this new State dictionary now enables our ExponentialBackoffRetryAttribute implementation as it existed. Note that if we weren't reusing the RandomizedBackoffRetryAttribute stateful class, this retry strategy could have been implemented without using State, since all required exponential backoff state can be determined/incremented solely based on the retry count which is maintained on the RetryContext.

So our addition of State is good since it adds flexibility for strategy implementors, but it'll be rare that people actually need to use it.